packages/client/src/MetricsPublisher.ts

```diff
@@ -79,8 +50,10 @@ export class MetricsPublisher {
     }
 
     private async publish(report: MetricsReport, streamId: string, partitionKey: string): Promise<void> {
+        await wait(Math.random() * this.config.maxPublishDelay)
```

Comment:
Since currently there is no way to abort a `wait`, it is footnote worthy to mention that we may attempt to publish after `publisher` has already been destroyed. Not sure if there is any practical way right now to avoid that from happening. (The idea of having a "singleton" factory from which we create all `wait` and `setTimeout` objects in the client would allow us to destroy all such pending tasks on "destroy").

Reply:
Yes, Publisher will throw if we call NetworkNodeFacade#publishToNode if a client is already destroyed (`NetworkNodeFacade:212`). It would be good to have a `abortController` support in `wait` so that client can exit immediatelly when `StreamrClient#destroy` is called...